<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets  
2.  Load the three datasets into three tables in a Db2 database
3.  Execute SQL queries to answer assignment questions 


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days. 

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

- <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv" target="_blank">Chicago Census Data</a>

- <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv" target="_blank">Chicago Public Schools</a>

- <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv" target="_blank">Chicago Crime Data</a>

**NOTE**: For the learners who are encountering issues with loading from .csv in DB2 on Firefox, you can download the .txt files and load the data with those:

- <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.txt" target="_blank">Chicago Census Data</a>
    
- <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.txt" target="_blank">Chicago Public Schools</a>
  
- <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.txt" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be stored in the database.

While it is easier to read the dataset into a Pandas dataframe and then PERSIST it into the database as we saw in Week 3 Lab 3, it results in mapping to default datatypes which may not be optimal for SQL querying. For example a long textual field may map to a CLOB instead of a VARCHAR. 

Therefore, **it is highly recommended to manually load the table using the database console LOAD tool, as indicated in Week 2 Lab 1 Part II**. The only difference with that lab is that in Step 5 of the instructions you will need to click on create "(+) New Table" and specify the name of the table you want to create and then click "Next". 

<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/LoadingData.png">

##### Now open the Db2 console, open the LOAD tool, Select / Drag the .CSV file for the first dataset, Next create a New Table, and then follow the steps on-screen instructions to load the data. Name the new tables as follows:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


### Connect to the database

Let us first load the SQL extension and establish a connection with the database

The following required modules are pre-installed in the Skills Network Labs environment. However if you run this notebook commands in a different Jupyter environment (e.g. Watson Studio or Ananconda) you may need to install these libraries by removing the `#` sign before `!pip` in the code cell below.


In [ ]:
# These libraries are pre-installed in SN Labs. If running in another environment please uncomment lines below to install them:
# !pip install --force-reinstall ibm_db==3.1.0 ibm_db_sa==0.3.3
# Ensure we don't load_ext with sqlalchemy>=1.4 (incompadible)
# !pip uninstall sqlalchemy==1.4 -y && pip install sqlalchemy==1.3.24
# !pip install ipython-sql

In [1]:
%load_ext sql

In the next cell enter your db2 connection string. Recall you created Service Credentials for your Db2 instance in first lab in Week 3. From your Db2 service credentials copy everything after db2:// (except the double quote at the end) and paste it in the cell below after ibm_db_sa://

<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/details.png">


In [2]:
# Remember the connection string is of the format:
# %sql ibm_db_sa://my-username:my-password@my-hostname:my-port/my-db-name?security=SSL
# Enter the connection string for your Db2 on Cloud database instance below
%sql ibm_db_sa://

(ibm_db_dbi.InterfaceError) ibm_db_dbi::InterfaceError: connect expects a not None dsn value
(Background on this error at: http://sqlalche.me/e/13/rvf5)
Connection info needed in SQLAlchemy format, example:
               postgresql://username:password@hostname/dbname
               or an existing connection: dict_keys([])


In [10]:
import pandas as pd

In [11]:
# Load CENSUS_DATA from ChicagoCensusData.csv
CENSUS_DATA = pd.read_csv('ChicagoCensusData.csv')
CENSUS_DATA.describe()

COMMUNITY_AREA_NUMBER  PERCENT_OF_HOUSING_CROWDED  \
count              77.000000                   78.000000   
mean               39.000000                    4.920513   
std                22.371857                    3.658981   
min                 1.000000                    0.300000   
25%                20.000000                    2.325000   
50%                39.000000                    3.850000   
75%                58.000000                    6.800000   
max                77.000000                   15.800000   

       PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
count                         78.000000                    78.000000   
mean                          21.739744                    15.341026   
std                           11.457231                     7.499497   
min                            3.300000                     4.700000   
25%                           13.350000                     9.200000   
50%                           19.050000                    13.850000   
75%                           29.150000                    20.000000   
max                           56.500000                    35.900000   

       PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
count                                     78.000000   
mean                                      20.330769   
std                                       11.746514   
min                                        2.500000   
25%                                       12.075000   
50%                                       18.650000   
75%                                       26.600000   
max                                       54.800000   

       PERCENT_AGED_UNDER_18_OR_OVER_64  PER_CAPITA_INCOME  HARDSHIP_INDEX  
count                         78.000000          78.000000       77.000000  
mean                          35.717949       25597.000000       49.506494  
std                            7.284421       15196.405541       28.690556  
min                           13.500000        8201.000000        1.000000  
25%                           32.150000       15804.750000       25.000000  
50%                           38.050000       21668.500000       50.000000  
75%                           40.500000       28715.750000       74.000000  
max                           51.500000       88669.000000       98.000000

In [12]:
# Load CHICAGO_PUBLIC_SCHOOLS from ChicagoPublicSchools.csv
CHICAGO_PUBLIC_SCHOOLS = pd.read_csv('ChicagoPublicSchools.csv')
CHICAGO_PUBLIC_SCHOOLS.describe()

School_ID      ZIP_Code  SAFETY_SCORE  Environment_Score  \
count     566.000000    566.000000    513.000000         513.000000   
mean   609681.791519  60629.727915     49.504873          47.766082   
std      8831.060946     20.248024     20.110837          16.215584   
min    400018.000000  60605.000000      1.000000           1.000000   
25%    609873.250000  60618.000000     35.000000          37.000000   
50%    610052.500000  60625.000000     48.000000          47.000000   
75%    610212.750000  60639.000000     61.000000          58.000000   
max    610544.000000  60827.000000     99.000000          99.000000   

       Instruction_Score  Rate_of_Misconducts__per_100_students_  \
count         513.000000                              566.000000   
mean           48.288499                               21.085866   
std            17.417176                               27.543835   
min             1.000000                                0.000000   
25%            37.000000                                4.525000   
50%            47.000000                               12.250000   
75%            59.000000                               26.975000   
max            99.000000                              251.600000   

       ISAT_Exceeding_Math__  ISAT_Exceeding_Reading__  ISAT_Value_Add_Math  \
count             476.000000                476.000000           468.000000   
mean               20.356303                 15.667017             0.068590   
std                16.850870                 15.208361             1.041174   
min                 0.000000                  0.000000            -3.500000   
25%                 9.175000                  6.100000            -0.600000   
50%                16.000000                 10.950000             0.000000   
75%                24.900000                 19.525000             0.700000   
max               100.000000                100.000000             3.600000   

       ISAT_Value_Add_Read  COLLEGE_ENROLLMENT  General_Services_Route  \
count           468.000000          566.000000              566.000000   
mean              0.022650          626.053004               39.084806   
std               1.042543          448.495105                6.165552   
min              -5.000000           21.000000               29.000000   
25%              -0.600000          342.250000               34.000000   
50%               0.000000          504.000000               39.000000   
75%               0.625000          790.500000               45.000000   
max               4.900000         4368.000000               49.000000   

       X_COORDINATE  Y_COORDINATE    Latitude   Longitude  \
count  5.660000e+02  5.660000e+02  566.000000  566.000000   
mean   1.163016e+06  1.885663e+06   41.841890  -87.677323   
std    1.605930e+04  3.285688e+04    0.090361    0.058390   
min    1.118114e+06  1.817242e+06   41.653674  -87.841052   
25%    1.152166e+06  1.860248e+06   41.771866  -87.717152   
50%    1.163553e+06  1.886889e+06   41.845521  -87.675878   
75%    1.174256e+06  1.911652e+06   41.913584  -87.636059   
max    1.202811e+06  1.950960e+06   42.021064  -87.533007   

       COMMUNITY_AREA_NUMBER        Ward  Police_District  
count             566.000000  566.000000       566.000000  
mean               37.706714   22.060071        11.710247  
std                21.584135   13.645574         6.688582  
min                 1.000000    1.000000         1.000000  
25%                23.000000   10.000000         7.000000  
50%                31.500000   21.000000        10.000000  
75%                58.000000   32.750000        17.000000  
max                77.000000   50.000000        25.000000

In [32]:
CHICAGO_PUBLIC_SCHOOLS.head()

School_ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School         Street_Address     City State  \
0                                 ES        615 W Kemper Pl  Chicago    IL   
1                                 ES  7511 S South Shore Dr  Chicago    IL   
2                                 ES     8010 S Kostner Ave  Chicago    IL   
3                                 ES     4619 S Wolcott Ave  Chicago    IL   
4                                 HS        3630 S Wells St  Chicago    IL   

   ZIP_Code    Phone_Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                                Link  \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network_Manager  ... Freshman_on_Track_Rate__  \
0        Fullerton Elementary Network  ...                      NDA   
1           Skyway Elementary Network  ...                      NDA   
2           Midway Elementary Network  ...                      NDA   
3         Pershing Elementary Network  ...                      NDA   
4  Southwest Side High School Network  ...                     91.8   

  X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
0  1171699.458  1915829.428  41.924497 -87.644522                     7   
1  1196129.985  1856209.466  41.760324 -87.556736                    43   
2  1148427.165  1851012.215  41.747111 -87.731702                    70   
3  1164504.290  1873959.199  41.809757 -87.672145                    61   
4  1175177.622  1880745.126  41.828146 -87.632794                    34   

  COMMUNITY_AREA_NAME  Ward Police_District                     Location  
0        LINCOLN PARK    43              18  (41.92449696, -87.64452163)  
1         SOUTH SHORE     7               4  (41.76032435, -87.55673627)  
2             ASHBURN    13               8  (41.74711093, -87.73170248)  
3            NEW CITY    20               9    (41.8097569, -87.6721446)  
4       ARMOUR SQUARE    11               9  (41.82814609, -87.63279369)  

[5 rows x 78 columns]

In [13]:
# Load CHICAGO_CRIME_DATA from ChicagoCrimeData.csv
CHICAGO_CRIME_DATA = pd.read_csv('ChicagoCrimeData.csv')
CHICAGO_CRIME_DATA.describe()

ID         BEAT    DISTRICT        WARD  \
count  5.330000e+02   533.000000  533.000000  490.000000   
mean   6.228301e+06  1187.037523   11.195122   22.608163   
std    3.027454e+06   660.592263    6.496631   13.080119   
min    2.114900e+04   111.000000    1.000000    1.000000   
25%    3.420516e+06   711.000000    6.000000   12.000000   
50%    6.231712e+06  1111.000000   10.000000   24.000000   
75%    8.875058e+06  1634.000000   16.000000   32.000000   
max    1.127717e+07  2535.000000   25.000000   50.000000   

       COMMUNITY_AREA_NUMBER  X_COORDINATE  Y_COORDINATE        YEAR  \
count             490.000000  5.290000e+02  5.290000e+02   533.00000   
mean               37.542857  1.162876e+06  1.886223e+06  2008.34334   
std                21.409514  1.650197e+04  3.009980e+04     4.94940   
min                 1.000000  1.100658e+06  1.814512e+06  2001.00000   
25%                23.000000  1.151715e+06  1.860454e+06  2004.00000   
50%                30.000000  1.163235e+06  1.891447e+06  2008.00000   
75%                58.000000  1.174618e+06  1.907711e+06  2012.00000   
max                77.000000  1.204126e+06  1.951001e+06  2018.00000   

         LATITUDE   LONGITUDE  
count  529.000000  529.000000  
mean    41.843429  -87.677818  
std      0.082798    0.060073  
min     41.645796  -87.905227  
25%     41.772556  -87.718166  
50%     41.857737  -87.676733  
75%     41.902298  -87.634677  
max     42.021178  -87.528223

In [14]:
%%capture
%sql sqlite://

In [15]:
%sql PERSIST CENSUS_DATA
%sql PERSIST CHICAGO_PUBLIC_SCHOOLS
%sql PERSIST CHICAGO_CRIME_DATA

 * sqlite://
 * sqlite://
 * sqlite://


'Persisted chicago_crime_data'

In [16]:
%%sql
SELECT COUNT(*) FROM CENSUS_DATA;


 * sqlite://
Done.


[(78,)]

In [17]:
%%sql
SELECT COUNT(*) FROM CHICAGO_PUBLIC_SCHOOLS;


 * sqlite://
Done.


[(566,)]

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [19]:
%%sql
SELECT COUNT(*) FROM CHICAGO_CRIME_DATA;

 * sqlite://
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [20]:
%%sql
SELECT "COMMUNITY_AREA_NAME"
FROM CENSUS_DATA
WHERE "PER_CAPITA_INCOME" < 11000;

 * sqlite://
Done.


[('West Garfield Park',),
 ('South Lawndale',),
 ('Fuller Park',),
 ('Riverdale',)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)



In [21]:
%%sql
SELECT "Case_Number"
FROM CHICAGO_CRIME_DATA
WHERE "Description" LIKE '%MINOR%';

 * sqlite://
Done.


[('HL266884',), ('HK238408',)]

### Problem 4

##### List all kidnapping crimes involving a child?


In [27]:
%%sql
SELECT "Case_Number","Description","Primary_Type"
FROM CHICAGO_CRIME_DATA
WHERE "Primary_Type" = 'KIDNAPPING' AND "Description" LIKE '%CHILD%';

 * sqlite://
Done.


[('HN144152', 'CHILD ABDUCTION/STRANGER', 'KIDNAPPING')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [30]:
%%sql
SELECT DISTINCT "Primary_Type"
FROM CHICAGO_CRIME_DATA
WHERE "Location_Description" LIKE '%SCHOOL%';

 * sqlite://
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the average safety score for each type of school.


In [39]:
%%sql
SELECT "Elementary, Middle, or High School" AS "School_Type", AVG("Safety_Score") AS "Average_Safety_Score"
FROM CHICAGO_PUBLIC_SCHOOLS
GROUP BY "Elementary, Middle, or High School"
ORDER BY "School_Type";

 * sqlite://
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line 


In [47]:
%%sql
SELECT "COMMUNITY_AREA_NUMBER","COMMUNITY_AREA_NAME", "PERCENT_HOUSEHOLDS_BELOW_POVERTY"
FROM CENSUS_DATA
ORDER BY "PERCENT_HOUSEHOLDS_BELOW_POVERTY" DESC
LIMIT 5;

 * sqlite://
Done.


[(54.0, 'Riverdale', 56.5),
 (37.0, 'Fuller Park', 51.2),
 (68.0, 'Englewood', 46.6),
 (29.0, 'North Lawndale', 43.1),
 (27.0, 'East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone?


In [46]:
%%sql
SELECT "COMMUNITY_AREA_NUMBER", COUNT(*) AS "TOTAL_CRIMES"
FROM CHICAGO_CRIME_DATA
GROUP BY "COMMUNITY_AREA_NUMBER"
ORDER BY "TOTAL_CRIMES" DESC
LIMIT 1;

 * sqlite://
Done.


[(25.0, 43)]

In [49]:
%%sql
SELECT
    CC."COMMUNITY_AREA_NUMBER",
    CC."COMMUNITY_AREA_NAME",
    COUNT(*) AS "TOTAL_CRIME"
FROM
    CHICAGO_CRIME_DATA CD
INNER JOIN
    CENSUS_DATA CC
ON
    CD."COMMUNITY_AREA_NUMBER" = CC."COMMUNITY_AREA_NUMBER"
GROUP BY
    CC."COMMUNITY_AREA_NUMBER", CC."COMMUNITY_AREA_NAME"
ORDER BY
    "TOTAL_CRIME" DESC
LIMIT 1;


 * sqlite://
Done.


[(25.0, 'Austin', 43)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [51]:
%%sql
SELECT "COMMUNITY_AREA_NAME"
FROM CENSUS_DATA
WHERE
    "HARDSHIP_INDEX" = (
        SELECT
            MAX("HARDSHIP_INDEX")
        FROM
            CENSUS_DATA
    );

 * sqlite://
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [52]:
%%sql
SELECT
    "COMMUNITY_AREA_NAME"
FROM
    CENSUS_DATA
WHERE
    "COMMUNITY_AREA_NUMBER" = (
        SELECT
            "COMMUNITY_AREA_NUMBER"
        FROM
            CHICAGO_CRIME_DATA
        GROUP BY
            "COMMUNITY_AREA_NUMBER"
        ORDER BY
            COUNT(*) DESC
        LIMIT 1
    );


 * sqlite://
Done.


[('Austin',)]

Copyright © 2020 [cognitiveclass.ai](cognitiveclass.ai?utm_source=bducopyrightlink&utm_medium=dswb&utm_campaign=bdu). This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork22-2022-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by    | Change Description                 |
| ---------- | ------- | ------------- | ---------------------------------- |
| 2021-11-17  | 2.6  | Lakshmi  | Updated library  |
| 2021-05-19 | 2.4     | Lakshmi Holla | Updated the question   |
| 2021-04-30 | 2.3     | Malika Singla | Updated the libraries   |
| 2021-01-15 | 2.2     | Rav Ahuja | Removed problem 11 and fixed changelog |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets |
| 2020-09-05 | 2.0     | Malika Singla | Moved lab to course repo in GitLab |
| 2018-07-18 | 1.0     | Rav Ahuja | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan | Created initial version |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
